pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py

```diff
@@ -253,28 +252,20 @@ def _log_gpus_metrics(self) -> None:
     Utilities and properties
     """
 
-    def on_new_batch(self, batch: Any) -> int:
-        # when the user requests `dataloader_iter`, we can't track the batch_size
-        # and this is left to user responsibility.
-        if not isinstance(batch, pl.utilities.fetching.StepFuncDataLoaderIter):
-            assert self.trainer._results is not None
-            return self.trainer._results.extract_batch_size(batch)
-        return 1
-
     def on_epoch_start(self) -> None:
         self._epoch_end_reached = False
 
-    def on_batch_start(self, batch_idx: int, batch: Any) -> int:
+    def on_batch_start(self, batch_idx: int, batch: Any) -> None:
         self._batch_idx = batch_idx
         self._epoch_end_reached = False
-        return self.on_new_batch(batch)
+        assert self.trainer._results is not None
+        self.trainer._results.current_batch = batch
```

Comment:
reviewer: we keep track of current batch instead of batch size since we are extracting it inside `self.log` call.

Reply:
Why not put that inside `on_new_batch`?

Reply:
if we do that and someone specifies `self.log(batch_size=...)` and uses an ambiguous batch, then it will raise a warning suggesting user to use `self.log(batch_size=...)` which seems odd. so we can't really extract the batch_size outside the `LightningModule.log` because we don't know until then whether the user specified the batch_size explicitly or not.

Reply:
also in the current implementation if someone is not even logging, they will get this warning because we try to extract the batch size regardless of the fact whether its required or not

Reply:
I believe this line should clean the previous batch_size.